docs/how-to-guides/upgrade-openebs.md

```diff
@@ -0,0 +1,248 @@
+OpenEBS control plane and data plane components work independently. Even if the
+control plane components are upgraded to `v2.2.0`, the data plane components
+continue to work with the older version `v1.12.0`.
+
+#### Upgrade cStor pools
+
+Get existing `StoragePoolClaims` by executing:
+
+```bash
+kubectl get spc
+NAME                           AGE
+cstor-pool-openebs-replica-3   2d21h26m
+```
+
+Create a Job to upgrade the existing cStor pools:
+
+```bash
+cat > upgrade-cstor-pools-1120-220.yaml <<EOF
+# upgrade-cstor-pools-1120-220.yaml
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  generateName: cstor-spc-1120220-
+  namespace: openebs
+spec:
+  backoffLimit: 4
+  template:
+    spec:
+      serviceAccountName: openebs-operator
+      containers:
+      - name:  upgrade
+        args:
+        - "cstor-spc"
+        - "--from-version=${OPENEBS_OLD_VERSION}"
+        - "--to-version=${OPENEBS_NEW_VERSION}"
+$(kubectl get cstorpools -n openebs --no-headers -o custom-columns=":metadata.name" | sed 's/.*/        - "&"/')
+        - "--v=4"
+        #DO NOT CHANGE BELOW PARAMETERS
+        env:
+        - name: OPENEBS_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        tty: true
+        image: openebs/m-upgrade:${OPENEBS_NEW_VERSION}
+        imagePullPolicy: Always
+      restartPolicy: OnFailure
+EOF
+```
+
+Apply the job to start the upgrade for cStor pools:
+
+```bash
+kubectl create -f upgrade-cstor-pools-1120-220.yaml
+```
+
+Ensure the job runs to completion.
+
+
+#### Upgrade cStor volumes
+
+Get existing `CStorVolumes` by executing:
+
+```bash
+k get cstorvolumes --all-namespaces
+NAMESPACE   NAME                                       STATUS    AGE     CAPACITY
+openebs     pvc-5b7d8ee3-59b8-4a02-bd6d-a4e66fbecf9f   Healthy   2d21h   50Gi
+openebs     pvc-e8f8b268-b81c-4201-841a-283f557c44a7   Healthy   2d21h   50Gi
+```
+
+
+Create a Kubernetes job to upgrade the existing cStor volumes:
+
+```bash
+cat > upgrade-cstor-vols-1120-220.yaml <<EOF
+# upgrade-cstor-vols-1120-220.yaml
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  generateName: cstor-vol-1120220-
+  namespace: openebs
+spec:
+  backoffLimit: 4
+  template:
+    spec:
+      serviceAccountName: openebs-operator
+      containers:
+      - name:  upgrade
+        args:
+        - "cstor-volume"
+        - "--from-version=${OPENEBS_OLD_VERSION}"
+        - "--to-version=${OPENEBS_NEW_VERSION}"
+$(kubectl get cstorvolumes -n openebs --no-headers -o custom-columns=":metadata.name" | sed 's/.*/        - "&"/')
+        - "--v=4"
+        #DO NOT CHANGE BELOW PARAMETERS
+        env:
+        - name: OPENEBS_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        tty: true
+        image: openebs/m-upgrade:${OPENEBS_NEW_VERSION}
+        imagePullPolicy: Always
+      restartPolicy: OnFailure
+EOF
+```
+
+```bash
+kubectl create -f upgrade-cstor-vols-1120-220.yaml
```

Comment:
Ditto versions and `kubectl apply`.

Reply:
https://github.com/kinvolk/lokomotive/pull/1163#discussion_r519635065